internal/cqarrow/inet.go

```diff
@@ -0,0 +1,182 @@
+package cqarrow
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+	"github.com/goccy/go-json"
+)
+
+type InetBuilder struct {
+	*array.ExtensionBuilder
+	dtype *InetType
+}
+
+func NewInetBuilder(mem memory.Allocator, dtype arrow.ExtensionType) *InetBuilder {
+	b := &InetBuilder{
+		ExtensionBuilder: array.NewExtensionBuilder(mem, dtype),
+		dtype:            dtype.(*InetType),
+	}
+	return b
+}
+
+func (b *InetBuilder) Append(v net.IPNet) {
+	b.ExtensionBuilder.Builder.(*array.StringBuilder).Append(v.String())
+}
+
+func (b *InetBuilder) UnsafeAppend(v net.IPNet) {
+	b.ExtensionBuilder.Builder.(*array.StringBuilder).UnsafeAppend([]byte(v.String()))
+}
+
+func (b *InetBuilder) AppendValues(v []net.IPNet, valid []bool) {
+	data := make([]string, len(v))
+	for i, v := range v {
+		data[i] = v.String()
+	}
+	b.ExtensionBuilder.Builder.(*array.StringBuilder).AppendValues(data, valid)
+}
+
+func (b *InetBuilder) UnmarshalOne(dec *json.Decoder) error {
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	var val net.IPNet
+	switch v := t.(type) {
+	case string:
+		_, data, err := net.ParseCIDR(v)
+		if err != nil {
+			return err
+		}
+		val = *data
+	case []byte:
+		_, data, err := net.ParseCIDR(string(v))
+		if err != nil {
+			return err
+		}
+		val = *data
+	case nil:
+		b.AppendNull()
+		return nil
+	default:
+		return &json.UnmarshalTypeError{
+			Value:  fmt.Sprint(t),
+			Type:   reflect.TypeOf([]byte{}),
+			Offset: dec.InputOffset(),
+			Struct: "String",
+		}
+	}
+
+	b.Append(val)
+	return nil
+}
+
+func (b *InetBuilder) Unmarshal(dec *json.Decoder) error {
+	for dec.More() {
+		if err := b.UnmarshalOne(dec); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *InetBuilder) UnmarshalJSON(data []byte) error {
+	dec := json.NewDecoder(bytes.NewReader(data))
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	if delim, ok := t.(json.Delim); !ok || delim != '[' {
+		return fmt.Errorf("fixed size binary builder must unpack from json array, found %s", delim)
+	}
+
+	return b.Unmarshal(dec)
+}
+
+// UUIDArray is a simple array which is a FixedSizeBinary(16)
```

Comment:
It looks like all the comments in this file that refer to `FixedSizeBinary` or `UUID` need to be updated (they were copied from UUID)